rlawt_mac.m

```diff
@@ -293,19 +296,30 @@ JNIEXPORT void JNICALL Java_net_runelite_rlawt_AWTContext_swapBuffers(JNIEnv *en
 	}
 
 	glFlush();
+
 	[(RLLayer*)ctx->layer performSelectorOnMainThread:
 		@selector(displayIOSurface:)
 		withObject: (id)(ctx->buffer[ctx->back])
 		waitUntilDone: true];
-	
+
 	ctx->back ^= 1;
 
 	if (!ctx->buffer[ctx->back]
-		|| IOSurfaceGetWidth(ctx->buffer[ctx->back]) != ctx->layer.frame.size.width 
-		|| IOSurfaceGetHeight(ctx->buffer[ctx->back]) != ctx->layer.frame.size.height) {
-		if (!rlawtCreateIOSurface(env, ctx)) {
-			return;
-		}
+		|| IOSurfaceGetWidth(ctx->buffer[ctx->back]) != ctx->layer.frame.size.width
+		|| IOSurfaceGetHeight(ctx->buffer[ctx->back]) != ctx->layer.frame.size.height
+		|| ctx->backingScaleFactor != ctx->layer.superlayer.contentsScale) {
```

Comment:
Since the scales are tied to the buffers I would probably make a `CGfloat scale[2]` in `AWTContext` to avoid this double-swap